src/commands/runPrompt.ts

```diff
@@ -168,42 +168,54 @@ export const runPrompt: (secrets: vscode.SecretStorage, mode: PromptOption) => v
         const ranges: Record<string, vscode.Range> = {};
         const getBaseFunctionRange = () => new vscode.Range(doc.lineCount, 0, doc.lineCount, 0);
         await spawnPromptImage(promptOption.id, runningLocal ? inputWorkspace! : workspaceFolder!.uri.fsPath, 'vscode-user', process.platform, async (json) => {
-            if (json.method === 'functions') {
-                const functions = json.params;
-                for (const func of functions) {
-                    const { id, function: { arguments: args } } = func;
-                    const params_str = args;
-                    let functionRange = ranges[id] || getBaseFunctionRange();
-                    if (functionRange.isSingleLine) {
-                        // Add function to the end of the file and update the range
-                        await writeToEditor(params_str);
-                        functionRange = new vscode.Range(functionRange.start.line, functionRange.start.character, doc.lineCount, 0);
+            switch (json.method) {
+                case 'functions':
+                    const functions = json.params;
+                    for (const func of functions) {
+                        const { id, function: { arguments: args } } = func;
+                        const params_str = args;
+                        let functionRange = ranges[id] || getBaseFunctionRange();
+                        if (functionRange.isSingleLine) {
+                            // Add function to the end of the file and update the range
+                            await writeToEditor(params_str);
+                            functionRange = new vscode.Range(functionRange.start.line, functionRange.start.character, doc.lineCount, 0);
+                        }
+                        else {
+                            // Replace existing function and update the range
+                            await writeToEditor(params_str, functionRange);
```

Comment:
Did you put these functions into code fences?

Reply:
TODO